ha/core/event_analyzer/parser/parser.py

```diff
@@ -176,36 +263,51 @@ def parse_event(self, msg: str) -> HealthEvent:
         try:
             message = json.dumps(ast.literal_eval(msg))
             cluster_resource_alert = json.loads(message)
-            cluster_id = Conf.get(const.HA_GLOBAL_INDEX, f"COMMON_CONFIG{_DELIM}cluster_id")
-            site_id = Conf.get(const.HA_GLOBAL_INDEX, f"COMMON_CONFIG{_DELIM}site_id")
-            rack_id = Conf.get(const.HA_GLOBAL_INDEX, f"COMMON_CONFIG{_DELIM}rack_id")
-            timestamp = str(int(time.time()))
-            event_id = timestamp + str(uuid.uuid4().hex)
-            node_id = cluster_resource_alert["_resource_name"]
-            resource_type = cluster_resource_alert["_resource_type"]
-            event_type = cluster_resource_alert["_event_type"]
-            timestamp = cluster_resource_alert["_timestamp"]
+            health_event_list = []
+            if self._conf_store is not None:
+                key_to_search = self.prepare_key(cluster_resource_alert)
+                # Log.error(key_to_search)
+                if self._conf_store.key_exists(key_to_search):
+                    sys_health_key_val, current_pod_restart_val, gen_id = \
+                       ClusterResourceParser._get_sys_health_key_val(key_to_search, self._conf_store)
+                    # Log.error(f'{cluster_resource_alert["_generation_id"]} {gen_id}')
+                    if cluster_resource_alert["_generation_id"] != gen_id:
+                        if current_pod_restart_val is not None and current_pod_restart_val:
+                            # If the incoming generation id doesn't match with already stored
+                            # value and assosciated pod_restart count is 1. That means,
+                            # its a pod restart case and this alert is already handled
+                            sys_health_key_val["events"][0]["specific_info"]["pod_restart"] = 0
```

Comment:
since a function is used to get current_pod_restart_val please use same / similar wrapper to set the same also

Reply:
Yes. Done